dbms/src/Interpreters/Join.cpp

```diff
@@ -269,42 +249,13 @@ size_t Join::getTotalByteCount() const
 
 void Join::setSampleBlock(const Block & block)
 {
-    std::unique_lock lock(rwlock);
+    //std::unique_lock lock(rwlock);
```

Comment:
Why do you comment this line? And why ``` std::unique_lock ``` was here? There are no condvars. ``` std::lock_quard``` fits this case better.

Reply:
It's possible to get multiple read locks in Join::joinBlock. The lock should not be exclusive there.
The line is commented cause it called from ctor only now (fucntion's been moved to private). If it would be called outside of ctor it must use this lock.

Reply:
I've added a comment.